modules/sdk-core/src/account-lib/mpc/tss/ecdsa/paillierproof.ts

```diff
@@ -0,0 +1,67 @@
+import { bitLength, randBits } from 'bigint-crypto-utils';
+import { bigIntFromBufferBE } from '../../util';
+import { gcd, modInv, modPow } from 'bigint-mod-arith';
+
+// Security parameters.
+const k = 128;
+const alpha = 2;
```

Comment:
According to https://eprint.iacr.org/2018/057.pdf#page6 section 5, we can pick $\alpha$ to be a value like 8009 and get down the number of values m in the proof to be 10 and still achieve statistical soundness of $2^{-128}$. There is a cost of added checks of divisibility of N by primes less than $\alpha$, but the paper has done a cost benefit analysis of changing alpha that we might benefit from considering.

Reply:
Note they have found the optimal $\alpha$ to be $319567$ which yields an $m = 7$ in our case.